ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -265,13 +267,14 @@ private static String idWatermark(CompactionInfo ci) {
   }
 
   /**
-   * @return true if any files were removed
+   * @return true if the cleaner has removed all files rendered obsolete by compaction
    */
   private boolean removeFiles(String location, ValidWriteIdList writeIdList, CompactionInfo ci)
       throws IOException, NoSuchObjectException, MetaException {
     Path locPath = new Path(location);
+    Map<Path, AcidUtils.HdfsDirSnapshot> dirSnapshots = null;
     AcidUtils.Directory dir = AcidUtils.getAcidState(locPath.getFileSystem(conf), locPath, conf, writeIdList, Ref.from(
-        false), false);
+        false), false, dirSnapshots);
```

Comment:
What's you expectation here? dirSnapshots would be always null.

Reply:
This will fill up the snapshot, so it can be used later

Reply:
how exactly?

Reply:
Oh, sorry. I was talking nonsense. Yes, we should not pass null in. I thought an empty map, but checked the code that would not get filled up either. There is a working example in AcidUtils.getAcidFilesForStats, you should create the snapshot beforehand